compiler/src/iree/compiler/InputConversion/Common/QuantizedConvToConv.cpp

```diff
@@ -0,0 +1,320 @@
+                                                        reassociationMap);
+
+      // Perform a summing operation across the kernel width and height.
+      auto poolTy = RankedTensorType::get(
+          {expandTy.getDimSize(0), newConvTy.getDimSize(1),
+           newConvTy.getDimSize(2), expandTy.getDimSize(3)},
+          accETy);
+
+      llvm::SmallVector<Value> poolDynDims;
+      if (expandTy.isDynamicDim(0))
+        poolDynDims.push_back(rewriter.create<tensor::DimOp>(loc, inputSum, 0));
+
+      if (newConvTy.isDynamicDim(1))
+        poolDynDims.push_back(rewriter.create<tensor::DimOp>(loc, newConv, 1));
+
+      if (newConvTy.isDynamicDim(2))
+        poolDynDims.push_back(rewriter.create<tensor::DimOp>(loc, newConv, 2));
+
+      if (expandTy.isDynamicDim(3))
+        poolDynDims.push_back(rewriter.create<tensor::DimOp>(loc, inputSum, 3));
+
+      Value poolTensor = rewriter.create<tensor::EmptyOp>(
+          loc, poolTy.getShape(), accETy, poolDynDims);
+
+      Attribute initialAttr = rewriter.getZeroAttr(accETy);
+      Value initialValue = rewriter.create<arith::ConstantOp>(loc, initialAttr);
+      poolTensor = rewriter
+                       .create<linalg::FillOp>(loc, ValueRange{initialValue},
+                                               ValueRange{poolTensor})
+                       .result();
+
+      llvm::SmallVector<int64_t> kernel{filterTy.getDimSize(0),
+                                        filterTy.getDimSize(1)};
+      llvm::SmallVector<Value> kernelDynDims;
+      for (int i = 0; i < 2; i++) {
+        if (filterTy.isDynamicDim(i))
+          kernelDynDims.push_back(
+              rewriter.create<tensor::DimOp>(loc, filter, i));
+      }
+
+      Value poolDims =
+          rewriter.create<tensor::EmptyOp>(loc, kernel, accETy, kernelDynDims);
+      inputSum =
+          rewriter
+              .create<linalg::PoolingNhwcSumOp>(loc, ArrayRef<Type>{poolTy},
+                                                ValueRange{inputSum, poolDims},
+                                                poolTensor, strides, dilations)
+              .getResult(0);
+
+      // Collapse the length-1 ending dimension away.
+      auto collapseTy =
+          RankedTensorType::get(poolTy.getShape().drop_back(), accETy);
+      inputSum = rewriter.create<tensor::CollapseShapeOp>(
+          loc, collapseTy, inputSum, reassociationMap);
+
+      // Apply the zero-point update based on the input sum.
+      newConv =
+          applyZeroPoint(rewriter, loc, newConv, inputSum, fZp, {0, 1, 2});
+    }
+
+    // Apply the final update that occurs when there are multiple zero-points.
+    if (!iZpIsZero && !fZpIsZero) {
+      Value dim0 = rewriter.create<tensor::DimOp>(loc, filter, 0);
+      Value dim1 = rewriter.create<tensor::DimOp>(loc, filter, 1);
+      Value dim2 = rewriter.create<tensor::DimOp>(loc, filter, 2);
+      Value mul1 = rewriter.create<arith::MulIOp>(loc, dim0, dim1);
```

Comment:
this code would feel easier to read with idiomatic names for local variables, instead of `mul1`, etc.